### ISSUE TITLE
✅ 🐞 Bug Report: Missing version string in included CSS widget-mega-menu.min.css with "Improved CSS Loading" (Related to #23790, #21879) [ED-15215] 

### ISSUE THREAD

**Issue by betatester07** [elementor/elementor#23821]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

I found out that the CSS file that contains base styling for the mega menu is printed in rendered HTML as `wp-content/plugins/elementor-pro/assets/css/widget-mega-menu.min.css` without any version string. Normally, query string with hash or version should be included, like `wp-content/plugins/elementor-pro/assets/css/widget-mega-menu.min.css?ver=3.16.3` to prevent issues with caching.

This is not directly a bug, but it is pretty much a standard procedure, all Wordpress style files are printed with a query string containing current version. Even other Elementor styling is included with query string, e.g. `wp-content/plugins/elementor-pro/assets/css/frontend.min.css?ver=3.16.1`. But for some reason, the mega menu CSS is included without any version string when "Improved CSS Loading" feature is enabled.

This is probably a cause for problems people have after update to 3.16 (#23753, #23813), because their browsers keep serving the old CSS file. Including query string with current Elementor version would definitely help to prevent this.

### Steps to reproduce

1) Have any kind of browser caching in place (instructing browsers to cache static files like .css)
2) Upgrade Elementor to 3.16 from older version
3) See broken mega menu

### Isolating the problem

- [ ] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [ ] I can reproduce this bug consistently following the steps above.

### System Info

<details>
<summary>Click to reveal</summary>

````txt



== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: mariadb.org binary distribution v10.5.17
	PHP Version: 8.1.23
	PHP Memory Limit: 512M
	PHP Max Input Vars: 5000
	PHP Max Post Size: 512M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.3.1
	Site URL:
	Home URL:
	WP Multisite: No
	Max Upload Size: 256 MB
	Memory limit: 512M
	Max Memory limit: 512M
	Permalink Structure: /%year%/%monthnum%/%day%/%postname%/
	Language: sk-SK
	Timezone: Europe/Bratislava
	Debug Mode: Inactive

== Theme ==
	Name: (Hello Elementor Child)
	Version: 2.0.0
	Author: Elementor Team
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 2.8.1
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: sk_SK

== Active Plugins ==
	Back In Stock Notifier for WooCommerce | WooCommerce Waitlist Pro
		Version: 4.0.1
		Author: codewoogeek

	Code Profiler
		Version: 1.6.4
		Author: Jerome Bruandet ~ NinTechNet Ltd.

	Complianz | GDPR/CCPA Cookie Consent
		Version: 6.5.3
		Author: Really Simple Plugins

	CTX Feed
		Version: 6.0.7
		Author: WebAppick

	DynamicConditions
		Version: 1.6.2
		Author: RTO GmbH

	Elementor
		Version: 3.16.3
		Author: Elementor.com

	Elementor Pro
		Version: 3.16.1
		Author: Elementor.com

	FiboSearch - AJAX Search for WooCommerce
		Version: 1.25.0
		Author: FiboSearch Team

	GTM4WP
		Version: 1.18.1
		Author: Thomas Geiger

	Perfect Brands WooCommerce
		Version: 3.2.3
		Author: QuadLayers

	Post SMTP
		Version: 2.6.0
		Author: Post SMTP

	Qi Addons For Elementor
		Version: 1.6.3
		Author: Qode Interactive

	Quatro splátkový predaj
		Version: 1.12.3
		Author: Eliaš Solutions s.r.o.

	Reviews and Rating - Google My Business
		Version: 4.24
		Author: Noah Hearle, Design Extreme

	Schema &amp; Structured Data for WP &amp; AMP
		Version: 1.20
		Author: Magazine3

	Sequential Order Numbers for WooCommerce
		Version: 1.10.1
		Author: SkyVerge

	TinyPNG - JPEG, PNG &amp; WebP image compression
		Version: 3.4.1
		Author: TinyPNG

	Toret Comgate
		Version: 2.3.2
		Author: toret.cz

	Toret GLS
		Version: 1.13.0
		Author: Toret.cz

	Toret Zásilkovna
		Version: 5.9.3
		Author: Toret.cz

	W3 Total Cache
		Version: 2.4.1
		Author: BoldGrid

	WooCommerce
		Version: 8.0.3
		Author: Automattic

	WooCommerce OBERON Connector
		Version: 1.0.14
		Author: WooConn.sk (Lapi Wowapi)

	Wordfence Security
		Version: 7.10.3
		Author: Wordfence

	WPForms Lite
		Version: 1.8.3.1
		Author: WPForms

	WPify Woo
		Version: 3.9.19
		Author: WPify s.r.o.

	YITH Color and Label Variations for WooCommerce
		Version: 1.34.0
		Author: YITH

	Yoast SEO
		Version: 21.1
		Author: Team Yoast


== Must-Use Plugins ==
	Code Profiler (mu-plugin)
		Version: 1.0
		Author: Jerome Bruandet


== Features ==
	Custom Fonts: 0
	Custom Icons: 1

== Integrácia ==
	
	woocommerce: Aktívny


== Experimenty so softvérom Elementor ==
	Optimalizovaný výstup DOM: Aktívne
	Vylepšené načítavanie prvkov: Aktívne
	Vylepšené načítavanie CSS: Neaktívne
	Ikony inline písma: Predvolené ako neaktívne
	Dodatočné prispôsobiteľné hraničné body: Neaktívne
	admin_menu_rearrangement: Predvolené ako neaktívne
	Kontajner Flexbox: Aktívne
	Upgrade Swiper Library: Aktívne
	Grid Container: Predvolené ako neaktívne
	Default to New Theme Builder: Predvolené ako aktívne
	Hello Theme Header &amp; Footer: Predvolené ako neaktívne
	Editor Top Bar: Aktívne
	Vstupné stránky: Neaktívne
	Nested Elements: Aktívne
	Obrázky pozadia Lazy Load: Predvolené ako neaktívne
	Pages Panel: Predvolené ako neaktívne
	Global Style Guide: Aktívne
	Page Transitions: Predvolené ako aktívne
	Notes: Neaktívne
	Form Submissions: Predvolené ako aktívne
	Scroll Snap: Predvolené ako aktívne
	Menu: Aktívne
	Taxonomy Filter: Predvolené ako neaktívne


== Záznam ==
	


== Elementor - Compatibility Tag ==
	
	Elementor Pro: Kompatibilné
	Qi Addons For Elementor: Nekompatibilné

== Elementor Pro - Compatibility Tag ==
	
	Qi Addons For Elementor: Nekompatibilné

````

</details>

**Comment by arunvel-01** [comment]
function load_custom_css() {
    wp_enqueue_style('mega-menu-css', plugin_dir_url( __FILE__ ) . 'path/to/your/mega-menu.css', array(), '3.16.3');
}

add_action('wp_enqueue_scripts', 'load_custom_css');
Replace 'path/to/your/mega-menu.css' with the actual path to the mega menu CSS file, and '3.16.3' with the current version of Elementor.

**Comment by nicholaszein** [comment]
Hello @betatester07! 

Thank you for submitting your issue! 🙏 

We are aware of this issue and our team is already working on a fix. We can’t say exactly when it'll be sorted out, but we kindly ask for your patience. Thanks for understanding! 😉

Meanwhile, a workaround is to **`Deactivate`** the **`Improved CSS Loading`** experimental feature if it hasn't been deactivated yet in your WordPress **`Dashboard`** > **`Elementor`** > **`Settings`** > **`Features`** page.

This feature has been deactivated by default in version **3.16.3** as a preemptive measure.

Kind regards

**Comment by betatester07** [comment]
> function load_custom_css() { wp_enqueue_style('mega-menu-css', plugin_dir_url( **FILE** ) . 'path/to/your/mega-menu.css', array(), '3.16.3'); }
> 
> add_action('wp_enqueue_scripts', 'load_custom_css'); Replace 'path/to/your/mega-menu.css' with the actual path to the mega menu CSS file, and '3.16.3' with the current version of Elementor.

good idea about temporary workaround! :+1: 


> Hello @betatester07!
> 
> Thank you for submitting your issue! 🙏
> 
> We are aware of this issue and our team is already working on a fix. We can’t say exactly when it'll be sorted out, but we kindly ask for your patience. Thanks for understanding! 😉
> 
> Meanwhile, a workaround is to **`Deactivate`** the **`Improved CSS Loading`** experimental feature if it hasn't been deactivated yet in your WordPress **`Dashboard`** > **`Elementor`** > **`Settings`** > **`Features`** page.
> 
> This feature has been deactivated by default in version **3.16.3** as a preemptive measure.
> 
> Kind regards

Thanks for the information.
Yes, disabling "Improved CSS Loading" fixes the issue (which we used as a workaround for now).

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Pro v3.23.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
